### PR TITLE
Changed how the options that were passed to mavproxy via the "-m" switch are parsed.

### PR DESCRIPTION
**Pull request by Jesse-Millwood**
Before it split up the string by spaces which didn't handle passing a string to "--cmd" or
using an "=" sign after the switch.
Now the string passed with the "-m" switch is split on spaces and "=" signs,
which is revaluated to reconstruct the split components which should be a single string by
looking for "\"" character pairs (denoting that that string should not be split), those
list items that are split up are joined back together as one list item to be passed on.
This has been tested with the call:
```
        sim_vehicle.py -m "--state-basedir /dir/dir/dir/ --cmd=\"param load $v2param; module load map\" --master=/dev/ttyUSB0 --console --map --out=udp:192.168.1.1:14550"
```
This would be nicer if regex could handle this but I could not construct a good regex pattern to not pick up the spaces within "\"" sings and still work properly.

### CODE REVIEW

File: Tools/autotest/sim_vehicle.py [hunk 1]
```diff
@@ -638,8 +638,40 @@ def start_mavproxy(opts, stuff):
     if "extra_mavlink_cmds" in stuff:
         extra_cmd += " " + stuff["extra_mavlink_cmds"]
 
+    # Parsing the arguments to pass to mavproxy
+    # Has been tested by calling:
+    # sim_vehicle.py -m "--state-basedir /dir/dir/dir/ --cmd=\"param load $v2param; module load map\" --master=/dev/ttyUSB0 --console --map --out=udp:192.168.1.1:14550"
```
Comment (khancyr) [inline Tools/autotest/sim_vehicle.py]: This comment should be remove in favor of only a descriptive comments : something like # Parsing the arguments to pass to mavproxy, split args on space and "=" signs something like that

Comment (Jesse-Millwood) [inline Tools/autotest/sim_vehicle.py]: @khancyr I changed that, thanks

**Comment by OXINARF** [comment]
@Jesse-Millwood Please read http://ardupilot.org/dev/docs/submitting-patches-back-to-master.html mainly the part about the commit message format.
Since this is your first contribution the fixes can be done by whoever merges this.

**Comment by Jesse-Millwood** [comment]
@OXINARF 
Sorry I completely missed that. To edit my commit message should I do that through the web interface here? Or is it something that some one else must do? I haven't contributed to a project before so I'm not as familiar with the work flow yet. 
Also would the subsystem be SITL?
Thanks

**Comment by OXINARF** [comment]
You can't edit through the web interface, you need to use Git command-line tool or a Git GUI. But for this one you can leave it for the person that eventually merges this to master.

This would be Tools, since this is inside the Tools folder. Could also be autotest, but I would prefer Tools since this script isn't only used for autotest.

P.S.: For anyone looking at the history of this PR, I changed the label to Tools and assigned to @peterbarker, GitHub is acting crazy.

**Comment by peterbarker** [comment]
@Jesse-Millwood Thanks for much for this.  It's certainly a wart that needs fixing.

I'm wondering if we can avoid writing our own parser for this, however.

```
#!/usr/bin/python

import shlex

foo = shlex.shlex("--state-basedir /dir/dir/dir/ --cmd=\"param load $v2param; module load map\" --master=/dev/ttyUSB0 --console --map --out=udp:192.168.1.1:14550", posix=True)
foo.wordchars += "-.:/"
foo = filter(lambda x : x != "=", foo)
for x in foo:
 print(x)
```

```
pbarker@bluebottle:~/rc/MAVProxy(arming-flags-fixes)$ ./test.py
--state-basedir
/dir/dir/dir/
--cmd
param load $v2param; module load map
--master
/dev/ttyUSB0
--console
--map
--out
udp:192.168.1.1:14550
pbarker@bluebottle:~/rc/MAVProxy(arming-flags-fixes)$ 
```

**Comment by Jesse-Millwood** [comment]
@peterbarker 
That `shlex` module is slick!
I never knew about that. I can make a new pull request with that `shlex` module if you'd like.

**Comment by peterbarker** [comment]
@Jesse-Millwood yeah, I think using a pre-existing parser might be a good idea.

Thanks!

**Comment by peterbarker** [comment]
@Jesse-Millwood nudge!

**Comment by peterbarker** [comment]
I've had a bit of a play with shlex today to see if it works.  It doesn't.

@Jesse-Millwood Are you still interested in pursuing this PR?

**Comment by Jesse-Millwood** [comment]
@peterbarker
Sorry for falling off the map
I am interested. I'll get to it this this weekend.

**Comment by peterbarker** [comment]
On Mon, 14 May 2018, Jesse Millwood wrote:

> @peterbarker
> Sorry for falling off the map
> I am interested. I'll get to it this this weekend.

Great!

As noted, I think shlex is a dead end, so some custom parsing here might 
be required (unless you can get optparse to play ball!).

You do need to allow for (and take advantage of) the fact that "'" is a 
valid quote character!

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**